luacompiler/src/lib/irgen/register_map.rs

```diff
@@ -1,43 +1,80 @@
-use std::{cell::RefCell, collections::HashMap};
+use std::{collections::HashMap, vec::Vec};
 
-/// Represents a structure that is used to keep track of the mapping
-/// between Lua variables and register ids.
-pub struct RegisterMap {
-    reg_count: RefCell<u8>,
-    reg_map: RefCell<HashMap<String, u8>>,
+/// Represents a tuple which is used to specify the lifetime of a register.
+/// For example if a register is first used by the 4th instruction of the bytecode, and
+/// used last by the 7th instruction, the register's lifetime would be (4, 8).
+pub struct Lifetime(usize, usize);
+
+impl Lifetime {
+    pub fn new(sp: usize) -> Lifetime {
+        Lifetime(sp, sp + 1)
```

Comment:
Why is it always `sp + 1`?

Reply:
When a register is first declared, its lifetime begins and ends at the same point. For instance:
```
1 ldi 1 2
2 ldi 2 2
3 add 3 1 2 // Register 3 has lifetime [3, 4)
```
This is because the register allocation algorithm that I will implement requires the end of lifetimes to be non-inclusive.

Reply:
OK.